Adds Carbonhell's Cheap Cuisine to vendors near you!

Kawaii-Big-Boss opened the pull request:
[Changelogs]: # (Your PR should contain a detailed changelog of notable changes, titled and categorized appropriately. This includes, new features, sprites, sounds, balance changes, admin tools, map edits, removals, big refactors, config changes, hosting changes and important fixes. An example changelog has been provided below for you to edit. If you need additional help, read https://github.com/tgstation/tgstation/wiki/Changelogs)

:cl: Kawaii-Big-Boss
add: Adds Carbonhell's Can of Cheap Cuisine, a mini microwave that makes food for you!
fix: fixes Carbonhell's ego being larger than I'd want
/:cl:

[why]: # (Please add a short description [two lines down] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.)
It's a goofy item that spices up vendors a bit. Some of the items it gives are more robust than you'd think at first glance (outside of the obvious "poisoning yourself/others to death") while also mocking a Headmin's cooking skills.

It's all banter, but you can't act like once you proudly display fucking _**ham discs**_ on the discord that someone wouldn't eventually mock you for it.

Thanks to McBawbaggings for helping me write the code.

fluxcapacitor1337 commented:
+1

Dudeinyourbackyard commented:
ham disc update is live

Carbonhell commented:
+1

Kawaii-Big-Boss commented:
code is good, +3 merge dis

McBawbaggings commented:
BBORN 2 LATE TO EXPLORE THE SEAS
BORN 2 EARLY TO EXPLOR THE STARS
BORN JUST IN TIME TO WITNESS HAM DIS

PR OF THE FUCKING YEARS BRO

McBawbaggings commented on hippiestation/code/game/objects/items/carbonhell.dm:
```diff
@@ -0,0 +1,112 @@
+	desc = "Tastes about as terrible as you'd expect."
+	icon = 'icons/obj/food/containers.dmi'
+	icon_state = "oliveoil"
+	list_reagents = list("amanitin" = 5, "soymilk" = 12, "atomicbomb" = 10, "hearty_punch" = 8) //guaranteed to fuck you up, soylent. also surprisingly robust if used before crit
```
wow!!

McBawbaggings commented on hippiestation/code/game/objects/items/carbonhell.dm:
```diff
@@ -0,0 +1,112 @@
+		return
+	new foodtype(get_turf(src))
+	used = TRUE
+	icon_state = "carboncan-open"
```
add an update 2 the descrpition here about how "It looks spent." or such

Kawaii-Big-Boss replied:
k

2Dollarbill commented:
+1, fugged'boutit

JohnGinnane commented on hippiestation.dme:
```diff
@@ -2553,6 +2553,7 @@
 #include "hippiestation\code\game\objects\effects\effect_system\effects_vapour.dm"
 #include "hippiestation\code\game\objects\items\apc_frame.dm"
 #include "hippiestation\code\game\objects\items\butterfly_knives.dm"
+#include "hippiestation\code\game\objects\items\carbonhell.dm"
```
Can you change the filename to something like "cheapcuisine.dm"? It doesn't seem right to name files after players..

praisenarsie replied:
can you not change the filename to something like "cheapcuisine.dm"? It's alright to name files after players.

praisenarsie replied:
ok and john and i talked and yeah you should change it to cheapcuisine.dm

Kawaii-Big-Boss replied:
donezo

JohnGinnane commented on hippiestation/code/game/objects/items/carbonhell.dm:
```diff
@@ -0,0 +1,113 @@
+	bitesize = 3
+	list_reagents = list("nutriment" = 4, "vitamin" = 2, "toxin" = 3, "fartium" = 5, "mushroomhallucinogen" = 3)
+	bonus_reagents = list("methamphetamine" = 2.5, "histamine" = 2)
+	tastes = list("pure, unadulterated misery" = 5)
```
How come there's a 5 here?

McBawbaggings replied:
that's how tastes work

JohnGinnane replied:
Yeah I did some checking and it looks like it's to give some weight to some tastes over others? e.g. for the jelly donut, "jelly" = 1 and "donut" = 3 so you're 3 times more likely to see "You can taste donut"? Should this list contain the names of reagents or can any fluff go here? If there's only one item then isn't the 5 redundant? Does this list combine with the reagents when forming tastes or does it override?

Kawaii-Big-Boss replied:
Overrides.

JohnGinnane commented on hippiestation/code/game/objects/items/carbonhell.dm:
```diff
@@ -0,0 +1,113 @@
+	bitesize = 1
+	list_reagents = list("nutriment" = 2, "vitamin" = 0.5, "toxin" = 1.5, "fartium" = 2)
+	bonus_reagents = list("methamphetamine" = 0.5, "histamine" = 0.5)
+	tastes = list("pure, unadulterated misery" = 2)
```
Same here?

Kawaii-Big-Boss replied:
Overrides.

JohnGinnane commented:
@Kawaii-Big-Boss did you test this?

dreamcrab commented:
lol.......................... +6

Kawaii-Big-Boss commented:
I renamed the file and checks are failing, good news.

Kawaii-Big-Boss commented:
Codecheck this now, it's almost been 24 hours and it's got +3

McBawbaggings commented:
What happens if you Destroy() the thing as it is cooking?
Does the sound loop runtime? I think the timer should be okay since https://github.com/HippieStation/HippieStation/blob/c58fb1bfc600b72bb52a6c69ffa3a25c39fdbee5/code/datums/callback.dm#L62
It checks to see if the object exists first before actually trying to call the proc. But I still remain unsure for the behaviour of the sound loop.

Kawaii-Big-Boss commented:
Haven't checked, but it's the same as the microwave code. Pretty sure it'll be okay.

Carbonhell commented on hippiestation/code/game/objects/items/cheapcuisine.dm:
```diff
@@ -0,0 +1,113 @@
+	soundloop = new(list(src), FALSE)
+
+/obj/item/cheapcuisine/attack_self(mob/user)
+	if(!used)
```
Why do you have both used and operating, can't you just use a single var?

Kawaii-Big-Boss replied:
Unfortunately, no. I use them both because it'd be cleaner.

Kawaii-Big-Boss replied:
ok

Carbonhell commented on hippiestation/code/game/objects/items/cheapcuisine.dm:
```diff
@@ -0,0 +1,113 @@
+/obj/item/cheapcuisine/proc/spawnFood(var/foodtype)
+	soundloop.stop()
+	operating = FALSE
+	if(isnull(foodtype))
```
Move this check in attack_self, nothing should run if you select Cancel in the input call.

Kawaii-Big-Boss replied:
You do realize that would require completely rewriting a proc, right?

Carbonhell commented on hippiestation/code/game/objects/items/cheapcuisine.dm:
```diff
@@ -0,0 +1,113 @@
+	new foodtype(get_turf(src))
+	used = TRUE
+	icon_state = "carboncan-open"
+	desc = "It's been used already."
```
Desc += not = or it will overwrite the normal desc

Kawaii-Big-Boss replied:
That's the point. Like how cans replace the desc with "This is rubbish"

Carbonhell commented on hippiestation/code/game/objects/items/cheapcuisine.dm:
```diff
@@ -0,0 +1,113 @@
+	list_reagents = list("nutriment" = 4, "vitamin" = 2, "toxin" = 3, "fartium" = 5, "mushroomhallucinogen" = 3)
+	bonus_reagents = list("methamphetamine" = 2.5, "histamine" = 2)
+	tastes = list("pure, unadulterated misery" = 5)
+	foodtype = GROSS | TOXIC // none of the food is actually real, it's just gross.
```
Corn potato is tasty you fuck

Kawaii-Big-Boss replied:
no

Carbonhell commented on hippiestation/code/game/objects/items/cheapcuisine.dm:
```diff
@@ -0,0 +1,113 @@
+	desc = "The jelly was so corrupted that it ended up gaining sentience."
+	list_reagents = list("nutriment" = 1, "vitamin" = 3, "amatoxin" = 7, "mushroomhallucinogen" = 3) // will KILL you
+	bonus_reagents = list("fartium" = 1, "coniine" = 3)
+	tastes = list("death" = 10)
```
Lmfao

Carbonhell commented on hippiestation/code/game/objects/items/cheapcuisine.dm:
```diff
@@ -0,0 +1,113 @@
+	name = "insta-jelly"
+	desc = "The jelly was so corrupted that it ended up gaining sentience."
+	list_reagents = list("nutriment" = 1, "vitamin" = 3, "amatoxin" = 7, "mushroomhallucinogen" = 3) // will KILL you
+	bonus_reagents = list("fartium" = 1, "coniine" = 3)
```
Coniine is a traitor only chem,with this you will make it available to normal people, are you sure about this choice

GuyonBroadway replied:
That's also six times a lethal dose.

Kawaii-Big-Boss replied:
Coniine is not a traitor only chem, and because it's in bonus_reagents it's not guaranteed to show up.
Coniine can show up in the Botanist's death berries, for instance. If you want me to lower the values, I can.

Carbonhell commented on hippiestation/code/game/objects/items/cheapcuisine.dm:
```diff
@@ -0,0 +1,113 @@
+	name = "butterdog"
+	desc = "This isn't a hot dog! It smells of heart disease!"
+	list_reagents = list("nutriment" = 4, "vitamin" = 2, "slimejelly" = 5, "mushroomhallucinogen" = 3)
+	bonus_reagents = list("impedrezene" = 3, "initropidril" = 1) // chance to have a heart attack and die
```
Initropidril too? Holy fuck wtf

Kawaii-Big-Boss replied:
Random chance, very small amount. Can remove.

Carbonhell commented on hippiestation/code/game/objects/items/cheapcuisine.dm:
```diff
@@ -0,0 +1,113 @@
+	desc = "The laziest food someone could possibly make, alongside some corn."
+	icon = 'hippiestation/icons/obj/food/food.dmi'
+	icon_state = "ham_disk"
+	list_reagents = list("nutriment" = 1, "vitamin" = 1, "soymilk" = 6)
```
>soymilk
What the fuck do you think i eat

Kawaii-Big-Boss replied:
I think you eat ham discs.

DaydreamPuppeh commented:
Merge this shit. Also I am pretty sure carbon also eats chef boyardee.

Pyko1 commented:
+1 concept is good, not sure about those 2 traitor chems tho, as someone is just going to just use them in mech syringegun to murderbone.

Kawaii-Big-Boss commented:
@Pyko1 Only initropidril is a traitor chem and I can swap it out if needed.

GuyonBroadway commented:
Coniine is traitor only jackass!

Kawaii-Big-Boss commented:
https://tgstation13.org/wiki/Guide_to_chemistry#Toxins

>death berries

Kawaii-Big-Boss commented:
Added Formaldehyde at the request of Guyon and Bantz in place of Coniine, replaced Initropidril with Sulfonal.
Still deadly, but not botany exclusive/traitor chems.
